compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -111,24 +125,29 @@ type HirType = crate::Type;
 pub fn monomorphize(
     main: node_interner::FuncId,
     interner: &mut NodeInterner,
+    force_unconstrained: bool,
 ) -> Result<Program, MonomorphizationError> {
-    monomorphize_debug(main, interner, &DebugInstrumenter::default())
+    monomorphize_debug(main, interner, &DebugInstrumenter::default(), force_unconstrained)
 }
 
 pub fn monomorphize_debug(
     main: node_interner::FuncId,
     interner: &mut NodeInterner,
     debug_instrumenter: &DebugInstrumenter,
+    force_unconstrained: bool,
 ) -> Result<Program, MonomorphizationError> {
     let debug_type_tracker = DebugTypeTracker::build_from_debug_instrumenter(debug_instrumenter);
     let mut monomorphizer = Monomorphizer::new(interner, debug_type_tracker);
+    monomorphizer.in_unconstrained_function = force_unconstrained;
     let function_sig = monomorphizer.compile_main(main)?;
 
     while !monomorphizer.queue.is_empty() {
-        let (next_fn_id, new_id, bindings, trait_method, location) =
+        let (next_fn_id, new_id, bindings, trait_method, is_unconstrained, location) =
             monomorphizer.queue.pop_front().unwrap();
         monomorphizer.locals.clear();
 
+        monomorphizer.in_unconstrained_function = is_unconstrained;
```

Comment:
```suggestion
        monomorphizer.in_unconstrained_function = is_unconstrained || force_unconstrained;
```

Reply:
Then I think we can leave this assert https://github.com/noir-lang/noir/blob/186fbb75c11fe575afd0d462f837331ccc307889/compiler/noirc_evaluator/src/ssa.rs#L279.

Reply:
Reading the pass some more it looks like this should not be necessary